pkg/printers/name_success_flags.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package printers
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// NameOrSuccessPrintFlags composes print flags capable of printing
+// a resource's fully-qualified Kind.group/name if the
+// output format matches a "name" printer, or a success
+// message indicating an action performed on a resource
+type NameOrSuccessPrintFlags struct {
+	*NamePrintFlags
+	*SuccessPrintFlags
```

Comment:
@soltysh @deads2k maybe it would be better to compose NamePrintFlags in SuccessPrintFlags? (the name printer is used in the success printer regardless)